meta-balena-common/recipes-containers/docker-disk/docker-disk.bb

```diff
@@ -55,6 +51,14 @@ do_compile () {
 	# mismatches
 	DOCKER=$(PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" which docker)
 
+        if [ -z "${BALENA_TOKEN}" ]; then
+                if [ -f "${HOME}/.balena/token" ]; then
+                    BALENA_TOKEN=$(cat ${HOME}/.balena/token)
+                else
+                    bbfatal "Please use \"balena login\" to login into balenaCloud for registry access"
```

Comment:
Is this going to be a new requirement for OS development?

Either set `BALENA_TOKEN` in our local.conf or make sure we are logged in on our host machine?

Reply:
Until public apps allow for unauthenticated pulls I am afraid so. This reminds me I should add a note to the README about it.

Reply:
@alexgg i believe public apps can be queried/pulled without authentication at the moment:

```
~
❯ curl -q --silent -H "Content-Type: application/json" "https://api.balena-cloud.com/v6/release?\$expand=release_tag,belongs_to__application,contains__image/image&\$filter=(belongs_to__application/any(a:a/app_name%20eq%20'amd64-supervisor'))%20and%20(release_tag/any(rt:(rt/tag_key%20eq%20'version')%20and%20(rt/value%20eq%20'v12.1.1')))" | jq -r "[.d[] | .contains__image[0].image[0] | [.is_stored_at__image_location, .content_hash] | \"\(.[0])@\(.[1])\"]"
[
  "registry2.balena-cloud.com/v2/62528bc04e71c603c8bc75505c19d815@sha256:f7d5593dd1be3a7972fff25b2d93b999defea64271557a3722f48bdf53123e35"
]

~ 10s
❯ docker pull registry2.balena-cloud.com/v2/62528bc04e71c603c8bc75505c19d815@sha256:f7d5593dd1be3a7972fff25b2d93b999defea64271557a3722f48bdf53123e35
sha256:f7d5593dd1be3a7972fff25b2d93b999defea64271557a3722f48bdf53123e35: Pulling from v2/62528bc04e71c603c8bc75505c19d815
aad63a933944: Pulling fs layer 
029d45fa2d66: Pulling fs layer 
f7224c33ec17: Pulling fs layer 
e99a32e3a1fb: Waiting 
9d1437470d9e: Waiting 
ed084d69f656: Waiting 
3aa1ca55dcd2: Waiting 
3a22a70f5295: Waiting 
7804309c8ea9: Waiting 
780442c4ed48: Waiting 
d2f33fce59f2: Waiting 
41f1f309c7ea: Pulling fs layer 
```

Is this not what you see?

Reply:
You're right @xginn8 . I was probably confused as for some time I wasn't able to properly flag apps as public. I will remove this requirement.